pkgs/games/minecraft/default.nix

```
@@ -41,16 +49,51 @@ let
     categories = [ "Game" ];
   };
 
-  envLibPath = lib.makeLibraryPath [
+  minecraft-launcher = stdenv.mkDerivation {
+    inherit pname version;
+
+    src = fetchzip {
+      url = "https://web.archive.org/web/20240711212151/https://launcher.mojang.com/download/Minecraft.tar.gz";
+      sha256 = "sha256-aCHY3yE/o8m7WbA+rY4mRd2FFJiqnBE5EtadwHLfbn0=";
+    };
+
+    icon = fetchurl {
+      url = "https://launcher.mojang.com/download/minecraft-launcher.svg";
+      sha256 = "0w8z21ml79kblv20wh5lz037g130pxkgs8ll9s3bi94zn2pbrhim";
+    };
+
+    nativeBuildInputs = [ copyDesktopItems ];
+
+    dontConfigure = true;
+    dontBuild = true;
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/bin
+      mv minecraft-launcher $out/bin
+
+      install -D $icon $out/share/icons/hicolor/symbolic/apps/minecraft-launcher.svg
+
+      runHook postInstall
+    '';
+
+    desktopItems = [ desktopItem ];
+  };
+in
+buildFHSEnv {
+  name = pname;
```

```suggestion
  name = "${pname}-${version}";
```

---

Which problem for example?

---

Keep it as it is. `buildFHSEnv` accepts a `version`.

```suggestion
    name = pname;
    inherit version;
```